options.py

```diff
@@ -291,6 +341,12 @@ class ButtonOption(BaseOption):
     _: KW_ONLY
     on_press: Callable[[Self], None] | None = None
 
+    def _to_json(self) -> JSON:
+        raise NotImplementedError("_to_json should never be called on a ButtonOption")
+
+    def _from_json(self, value: JSON) -> None:
+        raise NotImplementedError("_from_json should never be called on a ButtonOption")
```

Comment:
This is kind of awkward, if there's one specific type you always need to special case. You run into that problem further down in fact.

What if to json returned ellipsis, to mark it as unused, and from was just a noop?

Reply:
this is the main one, rest is nitpicks/consequences of this